topics/Estrategia_de_Qualidade.md

```diff
@@ -16,18 +16,222 @@ Pensar em estratégia de qualidade compreende:
+  iteração do usuário final com o sistema em questão. Deve seguir uma especificação
+  de teste (plano de teste, caso de teste ou cenário de teste) criada pela própria
+  equipe do projeto e baseada nos requisitos da aplicação.
+* **Testes automatizados de interface**: visam a automação de cenários de testes
+ a nível de tela, aonde simulam os comportamentos realizados manualmente por usuários.
+* **Testes automatizados de integração**: testam a interação entre sistemas e parte
+  isoladas de software afim de garantir que o conjunto se comporte da forma esperada
+  quando integrados.
+* **Testes exploratórios**: aqui a ideia é a mesma dos testes manuais, porém,
+  não existe uma especificação de teste a ser seguida.
+  Geralmente testes exploratórios são baseados em experiência,
+  uma vez que o testador deve explorar o sistema de forma a diversificar
+  ao máximo as opções de uso, encontrando assim falhas não identificadas nos testes
+  baseados em documentação.
+* **Testes de regressão**: visam garantir que outras partes do software não
+  tenham sido impactados por alterações recentes.
+  Tais testes podem ser manuais ou automatizados.
+* **Testes de fumaça**: são compostos por uma pequena suite de todos os testes do
+  sistema em questão, onde apenas as principais funcionalidades são validadas.
+  Tais testes podem ser manuais ou automatizados e
+  geralmente são executados após uma nova versão do software ser gerada.
+
+### Testes não funcionais
+
+Estes testes tem como finalidade verificar *como* o sistema
+se comporta mediante alguma situação. Alguns testes considerados não funcionais são:
+
+* **Testes de performance**: são testes que básicamente verificam como sua aplicação
+  se comporta em situações extremas, por exemplo,
+  quando vários usuários estão acessando a mesma página em um curto período de tempo.
+  Esses testes tem como objetivo captar possíveis pontos de gargalo no sistema.
+* **Teste de segurança**: visam garantir o quão seguro um sistema é,
+  expondo o mesmo a situações que simulam ataques que aproveitam vulnerabilidades
+  específicas, como por exemplo SQL Injection, Cross-Site Scripting e quebra de autenticação.
+* **Teste de usabilidade**: esses testes são usados para compreender melhor a interação
+  do usuário com o produto.
+  Geralmente é usado como uma técnica de avaliação,
+  onde existe um roteiro a ser seguido e um grupo de analistas observando e
+  coletando feedbacks ou problemas, como por exemplo,
+  ambiguidade de informações ou complicações no fluxo do sistema.
+
+### Recursos
+
+* [[Artigo] Tipos de testes de software](http://testesdesoftware.com/tipos-de-teste-de-software/)
+* [[Artigo] Os 13 principais tipos de testes de software](http://www.targettrust.com.br/blog/desenvolvimento/testes/os-13-principais-tipos-de-testes-de-software/)
+* [[Artigo] Performance testing in a Nutshell](https://www.thoughtworks.com/insights/blog/performance-testing-nutshell)
```

Comment:
faltou a indicação de língua inglesa: `:uk:`
```
* [[Artigo] Performance testing in a Nutshell](https://www.thoughtworks.com/insights/blog/performance-testing-nutshell) :uk:
```

* [[Artigo] Performance testing in a Nutshell](https://www.thoughtworks.com/insights/blog/performance-testing-nutshell) :uk: